### ISSUE TITLE
Easier "runtime/on-demand" pref change support

### ISSUE THREAD

**Issue by gregglind** [mozilla/web-ext#720]
### Is this a bug or feature request?

feature

### What is the current behavior?

The easiest way to get prefs into the firefox startup is to clone a profile and modify `prefs.js` or friends.  

### What is the expected or desired behavior?

```web-ext --prefs some.json|somePrefs.js  --prefs someOther.json``` which would be multicallable and applicable in right to left order.  Each of which is `json` or in the usual `userPrefs.js` sort of form.  

### justification

- Especially during development, having prefs control some aspects of  an addons behaviour is useful, such "theaddon.demoMode=true" or Logging prefs, or "don't phone home" and many other cases.
- it should be EASY to override "just a few" prefs, without having to go through all the hassle of profile creation and copying.

### Simplificiations / Alternatives

- only allow `json`

#### Prior art

- `cfx`, `jpm`
- `shield` does an enhanced form, where it has special shorthands for "the prefs tree for this addon"

**Comment by rpl** [comment]
@gregglind we recently introduced a similar feature in #658 : 
```
web-ext run --pref "a.custom.pref.name=value" --pref "othe.custom.pref.name=value"
```

it is not exactly the same as the one described above, mostly because it would be tedious to pass many customized preferences (but it could be easier when once we have added the support for a user-level and project-level config file as part of #176).

How it sounds to you?

If #658 and #176 are not enough to solve this issue, we will be absolutely more than happy to discuss about what is missing and how we can achieve it.
